### PR TITLE
Feature: Order Status

### PR DESCRIPTION
**Pull request by sgammon**
This changeset introduces methods and tools for updating a `bloombox.shop.order.Order` object, after it has been submitted to the server. Also introduced is `retrieve`, which will fetch an existing order from the server by ID.

Related issues:
- BLOOM-72 #comment Initial order status work.
- BLOOM-76 #comment Interface work for order status done.

Worklog:
- [x] Add method and structure to fetch orders
- [x] Add method to update an existing order object
- [x] Add new properties to the `bloombox.shop.order.Order` object
  - [x] Order ID
  - [x] Order Status
- [x] Add exported methods to get values from an order
  - [x] Retrieve order ID
  - [x] Retrieve order type
  - [x] Retrieve order status
- [x] Add local test for order status and retrieval
- [x] Temporary version bump -> `v1.0-beta21`
- [x] ~~Full property decoding for get-order~~ _NOTE: kicked_
- [x] Restore `1.0.0` version before merge

### CODE REVIEW

File: src/shop/ordering/orders.js [hunk 1]
```diff
@@ -86,6 +97,22 @@ bloombox.shop.order.Type = {
...
 
+/**
+ * Specifies statuses an order may be in.
+ *
+ * @enum {number}
+ * @export
+ */
+bloombox.shop.order.Status = {
+  'PENDING': proto.opencannabis.commerce.OrderStatus.PENDING,
+  'REJECTED': proto.opencannabis.commerce.OrderStatus.REJECTED,
```
Comment (sgammon) [inline src/shop/ordering/orders.js]: frontend-facing enum for order status is now available at `bloombox.shop.order.Status`

File: src/shop/ordering/orders.js [hunk 2]
```diff
@@ -207,16 +234,124 @@ bloombox.shop.order.Order = function Order(orderType,
...
+};
+
+
+/**
+ * Current status of this order.
+ *
+ * @return {bloombox.shop.order.Status}
+ * @export
+ */
+bloombox.shop.order.Order.prototype.getStatus = function() {
```
Comment (sgammon) [inline src/shop/ordering/orders.js]: ability to get the status of an order with `order.getStatus()`

File: src/shop/ordering/orders.js [hunk 3]
```diff
@@ -396,6 +545,278 @@ bloombox.shop.order.Order.prototype.addItem = function(item) {
...
+ * it accordingly.
+ *
+ * @param {bloombox.shop.OrderGetCallback} callback Callback to dispatch once
+ *        the order update has completed.
+ * @throws {bloombox.shop.order.OrderException} If no order ID is specified on
+ *        the current order, indicating it has not successfully been submitted
+ *        to the shop service yet.
+ * @export
+ */
+bloombox.shop.order.Order.prototype.update = function(callback) {
```
Comment (sgammon) [inline src/shop/ordering/orders.js]: method to update an order's data from the server:
`order.update(function(error, updatedOrder) { ... });`

File: src/shop/ordering/orders.js [hunk 4]
```diff
@@ -396,6 +545,278 @@ bloombox.shop.order.Order.prototype.addItem = function(item) {
...
+ *
+ * @param {string} key Key for the desired order.
+ * @param {bloombox.shop.OrderGetCallback} callback Callback to dispatch with
+ *        the resulting order.
+ * @throws {bloombox.shop.order.OrderException} If no order ID is specified on
+ *        the current order, indicating it has not successfully been submitted
+ *        to the shop service yet.
+ * @export
+ */
+bloombox.shop.order.Order.retrieve = function(key, callback) {
```
Comment (sgammon) [inline src/shop/ordering/orders.js]: you can also retrieve an order with:
`Order.retrieve("abc123", function(error, order) { ... });`

beware, for now orders retrieved this way only expose their ID, their type (`DELIVERY`/`PICKUP`) and their status (enumerated above in the new `bloombox.shop.order.Status`).

**Comment by noahdamiani** [comment]
Looks good.

**Pull request merged**